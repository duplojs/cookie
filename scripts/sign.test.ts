import { signCookie, verifyCookie } from "./sign";

describe("Test sign and verify cookie", () => {
	const secret = "itsVeryVeryVeryLongSecretKey秘密キーsecret2@&";

	it("should verify signed cookies correctly", () => {
		const value = "123456特殊な文字test2@é";

		const signed = signCookie(value, secret);
		const verified = verifyCookie(signed, secret);
		expect({
			signed,
			verified,
		}).toMatchSnapshot();
	});

	it("should handle invalid signature", () => {
		const value = "invalid.signature.signed";

		const result = verifyCookie(value, secret);
		expect({
			result,
		}).toMatchSnapshot();
	});

	it("should handle incomplete signature", () => {
		const value = "incomplete.signed";

		const result = verifyCookie(value, secret);
		expect({
			result,
		}).toMatchSnapshot();
	});

	it("should handle empty value", () => {
		const value = "";

		const result = verifyCookie(value, secret);
		expect({
			result,
		}).toMatchSnapshot();
	});

	it("should handle no dots", () => {
		const value = "no.dots.here";

		const result = verifyCookie(value, secret);
		expect({
			result,
		}).toMatchSnapshot();
	});
});
