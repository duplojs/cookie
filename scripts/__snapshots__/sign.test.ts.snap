// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test sign and verify cookie > should handle empty value 1`] = `
{
  "result": null,
}
`;

exports[`Test sign and verify cookie > should handle incomplete signature 1`] = `
{
  "result": null,
}
`;

exports[`Test sign and verify cookie > should handle invalid signature 1`] = `
{
  "result": null,
}
`;

exports[`Test sign and verify cookie > should handle no dots 1`] = `
{
  "result": null,
}
`;

exports[`Test sign and verify cookie > should verify signed cookies correctly 1`] = `
{
  "signed": "123456特殊な文字test2@é.5bcfea31e320c8efc941c94cbdbdf78ed69e74bd3e1214c4f2f19f5f33416769.signed",
  "verified": "123456特殊な文字test2@é",
}
`;
